src/phoenix/trace/llama_index/callback.py

```diff
@@ -18,21 +17,45 @@
 
 
 def _check_instrumentation_compatibility() -> bool:
-    if find_spec("llama_index") is None:
-        raise PackageNotFoundError("Missing `llama-index`. Install with `pip install llama-index`.")
-    # split the version string into a tuple of integers
-    llama_index_version_str = version("llama-index")
-    llama_index_version = tuple(map(int, llama_index_version_str.split(".")[:3]))
+    llama_index_version_str = None
+    try:
+        llama_index_version_str = version("llama-index")
+    except PackageNotFoundError:
+        pass
+
+    llama_index_core_version_str = None
+    try:
+        llama_index_core_version_str = version("llama-index-core")
+    except PackageNotFoundError:
+        pass
+
     instrumentation_version_str = version("openinference-instrumentation-llama-index")
-    instrumentation_version = tuple(map(int, instrumentation_version_str.split(".")[:3]))
-    # check if the llama_index version is compatible with the instrumentation version
+    instrumentation_version = _parse_semantic_version(instrumentation_version_str)
+
+    if llama_index_version_str is None:
+        if llama_index_core_version_str is None:
+            raise PackageNotFoundError(
+                "Missing `llama_index`. "
+                "Install with `pip install llama-index` or "
+                "`pip install llama-index-core` for a minimal installation."
+            )
+        if instrumentation_version < INSTRUMENTATION_MODERN_VERSION:
+            raise IncompatibleLibraryVersionError(
+                f"llama-index-core v{llama_index_core_version_str} is not compatible with "
+                f"openinference-instrumentation-llama-index v{instrumentation_version_str}. "
+                "Please upgrade openinference-instrumentation-llama-index to at least 1.0.0 via "
+                "`pip install 'openinference-instrumentation-llama-index>=1.0.0'`."
+            )
+        return True
```

Comment:
Explain this true - e.g. `-core` existence means that the instrumentation is modern "enough"

Reply:
This return statement corresponding to the case where only `llama-index-core` is installed and a modern version of our instrumentation is installed.